### PR TITLE
[CAP] Convenience methods for protobuf and some minor refactoring

### PR DESCRIPTION
**Pull request by rajan-chari**
## Why are these changes needed?
1) Sending protobufs requires repeating same boiler plate code
2) Support for non-color console logging
3) 

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: samples/apps/cap/py/pyproject.toml [hunk 1]
```diff
@@ -3,8 +3,8 @@ requires = ["hatchling"]
 build-backend = "hatchling.build"
 
 [project]
-name = "autogencap_rajan.jedi"
-version = "0.0.10"
+name = "autogencap"
```
Comment (thinkall) [inline samples/apps/cap/py/pyproject.toml:6]: You may want to update below doc as well:
 https://github.com/microsoft/autogen/blob/24d509c1b61c9df7ddc5dab095488636a1284441/samples/apps/cap/py/README.md?plain=1#L6

**Review by thinkall — COMMENTED** [review]
Thank you @rajan-chari ! I've just two tiny comments.

Below code:
https://github.com/microsoft/autogen/blob/24d509c1b61c9df7ddc5dab095488636a1284441/samples/apps/cap/py/demo/RemoteAGDemo.py#L12

`python demo/standalone/assistant.py` should be `python demo/standalone/Assistant.py`

**Review by qingyun-wu — APPROVED** [review]
Looks good! @rajan-chari It is good to go once Li's comment is addressed!

File: samples/apps/cap/py/pyproject.toml (hunk 1 above)
Comment (rajan-chari) [reply samples/apps/cap/py/pyproject.toml:6]: Thanks for the review!

**Pull request merged**